src/main/java/net/smartcosmos/cluster/auth/SmartCosmosAuthenticationProvider.java

```diff
@@ -227,4 +215,58 @@ private String getErrorResponseMessage(HttpStatusCodeException exception) {
         }
         return "";
     }
+
+    private SmartCosmosCachedUser checkedCachedUser(String username) {
+
+        if (users.containsKey(username)) {
+            final SmartCosmosCachedUser cachedUser = users.get(username);
+
+            if (System.currentTimeMillis() - cachedUser.getCachedDate()
+                .getTime() > cachedUserKeepAliveSecs * MILLISECS_PER_SEC) {
+                users.remove(username);
+            } else {
+                return cachedUser;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * This method is only utilized during the REFRESH Token phase and is designed only to see if the user account is still active.  No password is
+     * provided, because there was no password used by the client -- <b>only</b> the refresh token.
+     *
+     * @param username
+     * @return
+     * @throws UsernameNotFoundException
+     */
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+
+        log.debug("Checking to see if account {} is still active", username);
+
+        //        SmartCosmosCachedUser user = checkedCachedUser(username);
```

Comment:
Ha, I actually commented it out to force the load to be called, it should be uncommented or else it'll never use caching for refresh!